src/LinearMaps.jl

```diff
@@ -21,12 +21,40 @@ Base.size(A::LinearMap, n) = (n==1 || n==2 ? size(A)[n] : error("LinearMap objec
 Base.length(A::LinearMap) = size(A)[1] * size(A)[2]
 
 Base.:(*)(A::LinearMap, x::AbstractVector) = mul!(similar(x, promote_type(eltype(A), eltype(x)), size(A, 1)), A, x)
-function LinearAlgebra.mul!(y::AbstractVector, A::LinearMap, x::AbstractVector)
+function LinearAlgebra.mul!(y::AbstractVector, A::LinearMap, x::AbstractVector, α::Number=1, β::Number=0)
     length(y) == size(A, 1) || throw(DimensionMismatch("mul!"))
-    A_mul_B!(y, A, x)
+    if α == one(α)
+        β == zero(β) && (A_mul_B!(y, A, x); return y)
+        β == one(β) && (y .+= A * x; return y)
+        # β != 0, 1
+        rmul!(y, β)
+        y .+= A * x
+        return y
+    elseif α == zero(α)
+        β == zero(β) && (fill!(y, zero(eltype(y))); return y)
+        β == one(β) && return y
+        rmul!(y, β)
+        return y
+    else # α != 0, 1
+        rmul!(x, α)
```

Comment:
Right, I was too focused on avoiding allocations again. I'll replace the else-branch (alpha != 0, 1) by
```
β == zero(β) && (A_mul_B!(y, A, x); rmul!(y, α); return y)
β == one(β) && (y .+= rmul!(A * x, α); return y)
# β != 0, 1
rmul!(y, β)
y .+= rmul!(A * x, α)
return y
```
that seems allocation-free for the beta==0 case. Otherwise, we have a single allocation of the size of `y`.